apps/prairielearn/elements/pl-dataframe/pl-dataframe.py

```diff
@@ -174,9 +174,24 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     # Might be worth moving everything out of the CSS file and handle it all with the builtin styler.
     frame_style.set_table_attributes("class=pl-dataframe-table")
 
+    # Generate the initial HTML
+    frame_html_raw = frame_style.to_html()
```

Comment:
Nit: Is there a way to do this within the frame styler directly, instead of doing this after the fact? I ask because this might make this code more resilient to changes in Pandas. Not a huge deal either way. Might also be worth reporting upstream?

Reply:
I had looked into doing this, but I didn't see an easy way. https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.html

Reply:
+1, let's report this upstream before we merge this PR and include a link to the issue here.

Reply:
Filed an issue here: https://github.com/pandas-dev/pandas/issues/61350

Reply:
Looking at this again, it's not clear to me that this is needed at all. If one has `<tbody><tr><th>` (which Pandas does), that falls under the following from https://www.w3.org/WAI/WCAG21/Techniques/html/H63:

> For simple tables that have the headers in the first row or column, it is sufficient to simply use the th elements without scope.

Aside: your issue lacks a proper title.

Reply:
https://html-validate.org/rules/wcag/h63.html -- it looks like the implementation of this rule doesn't consider `tbody` or `thead`. Is that then a false positive?

Reply:
Filed https://gitlab.com/html-validate/html-validate/-/issues/303. I think this is a false positive.

Reply:
It sure seems like a false positive to me. I also ran the dataframe example question through https://ainspector.disability.illinois.edu/ on `master` and the first/third table didn't flag any problems with this. The second did, but only because headers are explicitly disabled there. Maybe `show-header="false"` needs a note about accessibility in the docs?

FWIW, we disable that rule in our other accessibility tests:

https://github.com/PrairieLearn/PrairieLearn/blob/7731a923ec81b86b7e225557d00910c6e9e89632/apps/prairielearn/src/tests/accessibility/index.test.ts#L101

Reply:
Also reverted back (no `scope="row"`) in 13aa8829c6ebef7d3587bdb0d7373f3aa81c135e.

Reply:
🙏 thank you! Going to mark this thread as resolved.